compliance_checker/cf/util.py

```diff
@@ -218,7 +218,8 @@ def get_coordinate(self, coord):
         if coord not in self.ds.variables:
             return
         if coord in self.reference_map:
-            self.reference_variables.add(self.obj.name)
+            if self.obj.name == coord:
+                self.reference_variables.add(self.obj.name)
```

Comment:
This fixes the graph recursive/self-referencing issue but it changes the nature of the test and how it reports. We no longer report variables that list coordinates that list themselves: we only list coordinates that are self referencing. Which in my opinion is MUCH better. Too many times people ask "why is temperature saying it is self referencing coordinates"... Well, it's not but latitude is...